_posts/2020-11-09-the-cloud-data-warehouse.md

```diff
@@ -0,0 +1,158 @@
+
+As an example, [Google BigQuery's slot system](https://cloud.google.com/bigquery/docs/slots) combines fixed units of compute (slot) with a variety of pricing models, such as:
+* On-demand pricing (pay for what you use, but you are limited to 2000 slots for each GCP project)
+* Fixed-rate pricing (pay to reserve a fixed amount of slots)
+* Flex pricing (pay to reserve a fixed amount of slots for a very specific time period (e.g. down to 60 seconds)
+
+The best part is that you can combine these pricing mechanisms to suit both your baseload and your mega-event loads.
+
+Snowflake takes a different approach with its [Virtual Warehouse](https://docs.snowflake.com/en/user-guide/warehouses-overview.html) concept, where you can 
+* scale vertically (increase the warehouse size to handle larger / more complex workloads)
+* scale horizontally (scale warehouse clusters in numbers to increase capacity for concurrency)
+* do both at the same time.
+
+Both platforms give you flexibility not only to choose the compute and pricing model that suits your workloads, but also to observe the current and historical workload profiles, helping you to match more closely the pricing model to the utilisation metrics. 
+
+## Which Cloud Data Warehouse should we choose?
+
+In general I think it is more productive to match a Cloud Data Warehouse to your intended usage and workloads, than it is to compare Cloud Data Warehouse offerings against each other. Typical early considerations include
+* What cloud platforms (if any) you currently work with
+* Where your data is coming from
+* Volume of historic data and growth rate of data
+* Which geographic regions you operate in
+* How many users / workloads you will typically support
+
+However, here are some hot takes I'd make for general analytical workloads based in Australia
+
+I would favour or at least consider **BigQuery** when any of the following apply
+* Most of the data came from existing workloads in the same Google Cloud Region
+* Data Warehouse workload was focused on data science or machine learning
+* There is a business requirement to stream data to the warehouse in near realtime
+
+I would consider **RedShift** when any of the following apply
+* I have a fairly constant and stable utilization profile that is well suited to pre-purchasing reserved instances for cost optimization
+* I want to to provide a unified data warehouse access layer comprised of federated AWS sources (e.g. operational RDS databases, S3) in order to build out native RedShift data marts
+
+**For most other use cases, I would favour Snowflake**.
+
+You may notice that we haven't discussed Amazon Redshift too much in our consideration of Cloud Data Warehouse solutions; we have and continue to work with customers using RedShift and are happy to continue doing so. Yet there are reasons why we often reach for other solutions:
```

Comment:
This is your guilt talking. 

I would drop this entire para in favour of something like...

"Redshift was one of the original Cloud Data Warehouse solutions and many customers are happy to continue using it."

Reply:
> "Redshift was one of the original Cloud Data Warehouse solutions and many customers are happy to continue using it."

That's wrong on both counts

Reply:
I disagree. I think it's important to call out why RedShift doesn't currently meet the definition of what we consider a cloud data warehouse to be. Instead it's a MPP database, and puts all of the burden of managing that lower level service on to the customer